File: w3/ex2/maxima.py
def find_local_maxima(arr):
    size = len(arr)

    if size < 2:
        return []

    maxima = []

    # Check first element
    if arr[0] > arr[1]:
        maxima.append(arr[0])

    # Check middle elements
    for i in range(1, size - 1):
        if arr[i] > arr[i - 1] and arr[i] > arr[i + 1]:
            maxima.append(arr[i])

    # Check last element
    if arr[size - 1] > arr[size - 2]:
        maxima.append(arr[size - 1])

    return maxima

arr = [2, 5, 7, 8, 4, 3, 5, 9, 4, 6, 8]
print(find_local_maxima(arr))

File: w3/ex2/maxima.c
#include <stdio.h>

void findLocalMaxima(int arr[], int size) {
    if (size < 2) return;

    // Check first element
    if (arr[0] > arr[1]) {
        printf("%d ", arr[0]);
    }

    // Check middle elements
    for (int i = 1; i < size - 1; i++) {
        if (arr[i] > arr[i - 1] && arr[i] > arr[i + 1]) {
            printf("%d ", arr[i]);
        }
    }

    // Check last element
    if (arr[size - 1] > arr[size - 2]) {
        printf("%d", arr[size - 1]);
    }
}

int main() {
    int arr[] = {2, 5, 7, 8, 4, 3, 5, 9, 4, 6, 8};
    int size = sizeof(arr) / sizeof(arr[0]);

    findLocalMaxima(arr, size);
    return 0;
}
